sys/include/net/gnrc/netif/mac.h

```diff
@@ -86,6 +86,13 @@ typedef struct {
      */
     gnrc_lwmac_t lwmac;
 #endif
+
+#ifdef MODULE_GNRC_GOMACH
+    /**
+     * @brief GoMacH specific structure object for storing GoMacH internal states.
+     */
+    gnrc_gomach_t gomach;
```

Comment:
Since I doubt you can run two MAC protocols on the same interface: Maybe make those a union with `lwmac` above.

```C
    union {
#ifdef MODULE_GNRC_LWMAC
        lwmac;
#endif
#ifdef MODULE_GNRC_GOMACH
        gomach;
#endif
    } prot;
```

This is then accessible via `netif->mac.prot.lwmac` and `netif->mac.prot.gomach` respectively.

Reply:
But, we don't want the same interface to run two (or more) MACs at the same time, right? This will leads to chaos for managing the same radio.

Reply:
> But, we don't want the same interface to run two (or more) MACs at the same time, right? This will leads to chaos for managing the same radio.

Yes, but we might want to have two (or more) interfaces running two (or more) MAC protocols at the same time. If that is the case, your approach would be very wasteful, since every interface would need to allocate memory for every MAC protocol. With the solution I proposed each interface only requires the space of the largest one.

Reply:
OK, I see it. But, on the other hand, this maybe another issue that related to both  LWMAC and GoMacH (and other upcoming MACs). So, I would like to handle that in a new specific PR, what do you think? :-)

Reply:
Mh, but since this change is introduce a second MAC, I really would prefer to do it right away...

Reply:
OK, will do it~

Reply:
By the way, as far as I understand the usage of `union`, if we want to run two MACs for two radios at the same time, we do want to separate the parameters storage of them, right? Putting the two MACs in the union, namely, both GoMacH and LwMAC sharing the same piece of memory, won't the operations of GoMacH will also unexpectedly affect the parameters of LwMAC, which cause chaos?

Reply:
Well they are separated, since they are on different netif structs, right? E.g. radio 0 would use `netifs[0].mac.prot.gomach` and radio 1 would use `netifs[1].mac.prot.lwmac` and since you usually know which protocol you are using (because you are in the code of the according protocol which already receives the interface you are on) you wouldn't mix-up the interfaces.

Reply:
=> no chaos, but well designed optimization.

Reply:
OK!~ Thanks!~ Update immediately.

Reply:
Addressed!~